internal/backends/postgresql/collection.go

```diff
@@ -312,42 +312,45 @@ func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams
 
 // Stats implements backends.Collection interface.
 func (c *collection) Stats(ctx context.Context, params *backends.CollectionStatsParams) (*backends.CollectionStatsResult, error) {
-	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
+	if params.Refresh {
+		p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
 
-	if p == nil {
-		return nil, backends.NewError(
-			backends.ErrorCodeCollectionDoesNotExist,
-			lazyerrors.Errorf("no ns %s.%s", c.dbName, c.name),
-		)
-	}
+		if p == nil {
+			return nil, backends.NewError(
+				backends.ErrorCodeCollectionDoesNotExist,
+				lazyerrors.Errorf("no ns %s.%s", c.dbName, c.name),
+			)
+		}
 
-	coll, err := c.r.CollectionGet(ctx, c.dbName, c.name)
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
+		coll, err := c.r.CollectionGet(ctx, c.dbName, c.name)
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
 
-	if coll == nil {
-		return nil, backends.NewError(
-			backends.ErrorCodeCollectionDoesNotExist,
-			lazyerrors.Errorf("no ns %s.%s", c.dbName, c.name),
-		)
-	}
+		if coll == nil {
+			return nil, backends.NewError(
+				backends.ErrorCodeCollectionDoesNotExist,
+				lazyerrors.Errorf("no ns %s.%s", c.dbName, c.name),
+			)
+		}
 
-	stats, err := collectionsStats(ctx, p, c.dbName, []*metadata.Collection{coll})
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
+		stats, err := collectionsStats(ctx, p, c.dbName, []*metadata.Collection{coll})
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
 
-	return &backends.CollectionStatsResult{
-		CountObjects:   stats.countRows,
-		CountIndexes:   stats.countIndexes,
-		SizeTotal:      stats.sizeTables + stats.sizeIndexes,
-		SizeIndexes:    stats.sizeIndexes,
-		SizeCollection: stats.sizeTables,
-	}, nil
+		return &backends.CollectionStatsResult{
+			CountObjects:   stats.countRows,
+			CountIndexes:   stats.countIndexes,
+			SizeTotal:      stats.sizeTables + stats.sizeIndexes,
+			SizeIndexes:    stats.sizeIndexes,
+			SizeCollection: stats.sizeTables,
+		}, nil
+	}
+	return &backends.CollectionStatsResult{}, nil
```

Comment:
`Refresh: false` does not mean "do nothing and return empty structure"

Reply:
I apologise for the mistake, I'll make changes.